Disable email notifications during order import

nagpai opened the pull request:
Fixes #4 

#### Summary 
The PR disables all WP email notifications, including Woo, while importing new orders and creating new customers. The PR uses the `pre_wp_mail` hook  

#### Testing instructions
* Best tested on a local WP site  and with our test shop credentials. LocalWP has a MailHog app that captures all emails being sent out and received successfully. 
* Checkout to trunk ( without the changes ) and do a test `wp migrator orders` .  Within localWP browse to `Tools > MailHog > Open MailHog`

![image](https://github.com/woocommerce/migrator-cli/assets/4162931/4c533ebc-66e9-4e24-b7de-0d4b9463247b)

* You will see a trail of emails sent by the import operation
* Empty the MailHog mailbox
* Now check-out to this branch, and then repeat the steps
* You should see an empty MailHog Mailbox. 

#### Additional testing for `--send-notifications` flag added in 4d6fea0 ( Props @leonardola )

* Empty existing test content ( users, orders, mail logs etc )
* Run `wp migrator orders --send-notifications` 
* Check MailHog logs, where you should see attempts to email `New orders` email to admin, and `New Account created` mails to masked emails of users. 
* Repeat the test with `wp migrator orders --send-notifications --mode=live` 
* You will see send attempts to actual email address in MailHog

Commit:
Initial chanegs

Commit:
small changes

Commit:
Merge changes from Jumble contacts PR branch

Commit:
minor

Commit:
Disable emails during migration.

Commit:
Merge branch 'trunk' into add/disable-wp-mail

leonardola reviewed (commented):
Looks good to me. We'll need to update the README to let users know emails won't be sent when running the tool. It might be good to have a flag to allow sending emails when the users want to.

Commit:
Add flag for enabling email notifications

nagpai commented:
@leonardola . Thanks a lot for the review and the suggestion. I have updated the README, and added a flag to include notifications with warning + confirmation. Request your eyes to make sure I have done it correctly. Thanks! 
Commit - 4d6fea0

leonardola reviewed (approved):
Looks good now. Thanks for adding this feature
